ecl/eclagent/eclagent.cpp

```diff
@@ -1949,15 +1932,17 @@ void EclAgent::doProcess()
     // signal to any lingering Thor's that job is complete and they can quit before timeout.
     executeGraphOnLingeringThor(*wuRead, nullptr, nullptr);
 #endif
-
+    while (clusterNames.ordinality())
```

Comment:
This should probably be within the try {} code.  Would also be worth commenting why it is happening before w is initialised.